extensions/amp-story/0.1/amp-story-page.js

```diff
@@ -138,6 +149,20 @@ export class AmpStoryPage extends AMP.BaseElement {
   }
 
 
+  /** @private */
+  initializeMediaPool_() {
+    const storyEl = dev().assertElement(
+        closestBySelector(this.element, 'amp-story'),
+        'amp-story-page must be a descendant of amp-story.');
+
+    storyEl.getImpl()
+        .then(storyImpl => {
+          this.mediaPoolResolveFn_(MediaPool.for(storyImpl));
```

Comment:
IIUC, the `amp-story-page` creates a `MediaPool` on behalf of its parent `amp-story` and waits for it to complete? 

Why not create this `MediaPool` in `amp-story` and signal its children instead?

Reply:
It's a singleton created per root element (which, in this case, the story is the root, but we are envisioning other cases as well).  We can pass it from `amp-story`, but we were actually doing this before and it gets kind of ugly.  It's especially bad if/when you need to pass it through multiple layers of children...